--- /usr/lib/python3.9/site-packages/paramiko/transport.py      2025-08-07 15:09:40.000000000 +0200
+++ /usr/lib/python3.9/site-packages/paramiko/transport.py.quieter      2025-08-25 13:34:22.303856548 +0200
@@ -2279,7 +2279,9 @@
                         "server" if self.server_mode else "client", e
                     ),
                 )
-                self._log(ERROR, util.tb_strings())
+                # NOTE: we degraded this noisy traceback from ERROR to INFO
+                #       It is usually just ssh vulnerability scans failing.
+                self._log(INFO, util.tb_strings())
                 self.saved_exception = e
             except EOFError as e:
                 self._log(DEBUG, "EOF in transport thread")
